Enhancement: Roadmap page for the Overview section

rishabhknowss opened the pull request:
<!-- In order to keep off topic discussion to a minimum, it helps if the "work to be done" is already agreed on. -->
<!-- Ideally, a GitHub Issue with the label `Status: Consensus` will have been concluded, and linked to in this PR. -->
<!--
Thanks for submitting a pull request! Please provide enough information so that others can review your pull request.
-->

**What kind of change does this PR introduce?**

Adds a Roadmap page

**Issue Number:**
<!-- Pick one of the below options.  Please remove those which don't apply. -->
-  Closes #181  <!-- Replace ___ with the issue number this PR resolves -->



**Screenshots/videos:**

<!--Add screenshots or videos wherever possible.-->
![image](https://github.com/user-attachments/assets/50231616-6580-4e2b-a30e-2378f4a3f163)

<!--Add link to it-->

rishabhknowss commented:
i think i forgot to delete my env files ( but it should get ignore with git :')  )

benjagm commented:
Hi @rishabhknowss. is this ready for review? It seems uncompleted. One thing to consider is remember that we have dark/light theme selection so you need to implement that behavior as well. I am saying this because it seems that the current version in this PR was created only for dark theme.  

In addition, the roadmap page will be into the overview section of the docs, so my advice is to place this feature into the /pages/overview/roadmap folder, and use the same tsx approach than other overview pages like case studies or use cases.

rishabhknowss commented:
ohh, totally forgot about the themes....making changes

benjagm commented:
> ohh, totally forgot about the themes....making changes

Please, check the correct colors in all labels and styles of each card because in both dark and light themes.

rishabhknowss commented:
I have added both the themes , fixed the route , adjusted the labels according to the [board]( https://github.com/orgs/json-schema-org/projects/14/views/1) and made it responsive 
Suggest more changes :) 

[Screencast from 2024-10-13 15-20-01.webm](https://github.com/user-attachments/assets/788a1bfd-8756-4bd9-ba8a-5ec7b4be28bd)

gregsdennis commented:
@benjagm I think we may want to consider revisiting the roadmap given our changed circumstances.  Have our goals or priorities changed since we're not working on this full time anymore?

Probably a good idea to figure this out before we post it on the website.

benjagm commented:
> @benjagm I think we may want to consider revisiting the roadmap given our changed circumstances. Have our goals or priorities changed since we're not working on this full time anymore?
> 
> Probably a good idea to figure this out before we post it on the website.

This is a great point. The great part of this approach is the data is coming from this board: https://github.com/orgs/json-schema-org/projects/14 so it will dynamically show live data from GitHub. While the roadmap discussion is in progress we can hide the table and show a different message explaining that the roadmap discussion is happening and inviting everyone to be part of it.

rishabhknowss commented:
yup guys it will reflect the changes on the roadmap board

benjagm reviewed (changes requested):
Are you sure this is the final version of the code? I am missing most of the UX you shared in this screenshot:
![375988954-50231616-6580-4e2b-a30e-2378f4a3f163](https://github.com/user-attachments/assets/3056f729-fa92-40aa-b84b-9fc08e121cae)

This is what I see in this version:
<img width="1508" alt="Screenshot 2024-10-15 at 09 23 31" src="https://github.com/user-attachments/assets/0de5bdd1-cd0a-40a2-a10c-72ba6a2b1fb6">

In addition, the roadmap page will be part of the sidebar in the overview section of our docs. You should use a different Typescript page with sidebar capabilities. Please use the same approach used for other pages inside the overview section:
https://github.com/json-schema-org/website/blob/main/pages/overview/use-cases/index.page.tsx

or:

https://github.com/json-schema-org/website/blob/main/pages/overview/case-studies/index.page.tsx

rishabhknowss commented:
okay resolving the issue

rishabhknowss commented:
Hi @benjagm , after following the same approach as other pages have, the design aspect has changed a little bit. Should I push the same, or am I thinking of keeping only 1 column instead of two alternate cards?


the earlier design 
![image](https://github.com/user-attachments/assets/dd684480-c51d-4467-80d0-0c8ebcc3c7c0)



new design ( imo this is better )
![image](https://github.com/user-attachments/assets/cbfa7c73-fbc6-46af-9e9f-15ee649731b2)

rishabhknowss commented:
Hi @benjagm , thanks for the follow-up. I think the code is now ready for review. I have changed the design a little bit and followed the same approach as other pages have.

![image](https://github.com/user-attachments/assets/4ea41b5f-8214-40eb-b736-5d7bdce0bc97)

rishabhknowss commented:
I think there's still a deployment issue, as I can also see no UX. However, it is working perfectly on local. 
i am debugging the issue

rishabhknowss commented:
i have implemented something , let me see if it works

rishabhknowss commented:
I also served the website locally, and it is working fine. I think there might be another issue

![image](https://github.com/user-attachments/assets/b90e1c37-25d3-46c4-861d-76989ff6f02b)

benjagm commented:
It seems that Tailwind’s PurgeCSS, because of the theme definition, is affecting the styles you are trying to use.

To avoid this, you can add a "safelist" in the `tailwind.config.js` file to ensure that these classes aren’t purged.

````
module.exports = {
  content: [
    './pages/**/*.{js,ts,jsx,tsx,md}',
    './components/**/*.{js,ts,jsx,tsx,md}',
  ],
  darkMode: "class",
  safelist: [
    'bg-green-600',
    'bg-yellow-600',
    'bg-emerald-600',
    'bg-purple-600',
    'bg-pink-600',
    'bg-orange-600',
    'bg-gray-600',
    'dark:bg-yellow-500',
    'dark:bg-purple-500',
    'dark:bg-green-500',
    'dark:bg-blue-500',
    'dark:bg-pink-500',
    'dark:bg-gray-500',
    // Add any other classes you dynamically generate
  ],
  theme: {
    screens: {
  ````

rishabhknowss commented:
Thanks @benjagm  , its working now

benjagm reviewed (changes requested):
This is looking great and we just need to make some UX adjustments:

<img width="206" alt="Screenshot 2024-10-18 at 17 53 29" src="https://github.com/user-attachments/assets/47abf1e9-34c4-44fb-b1d2-7df4cf5d8be3">
* Those gradients remind me too much AsyncAPI's branding. Can we change it to use our cooperative blue as base color instead of that violet?

* We need some separation between the circle attached to each card and the vertical line.

* Let's remove the names of the assigned contributors.

* Let's define consistent colors for all the labels:
  Effort: Red for very high colors, to yellow to medium values to green to low values
  Impact: Green for very high colors, to yellow to medium values to red to low values
  Status: Green in progress, violet completed, red paused, blue deferred and planned

I hope this helps.

rishabhknowss commented:
making changes

benjagm commented on components/Sidebar.tsx:
```diff
@@ -109,6 +109,7 @@ const getDocsPath = [
   '/overview/use-cases',
   '/overview/code-of-conduct',
   '/overview/faq',
+  'overview/roadmap',
```
This path is missing the / characther in the beginning. Please use '/overview/roadmap',

```suggestion
  '/overview/roadmap',
```

benjagm commented on pages/overview/roadmap/index.page.tsx:
```diff
@@ -0,0 +1,128 @@
+export default function Roadmap() {
+  const newTitle = 'Roadmap';
+  const markdownFile = '_indexPage';
+  const date = new Date().getFullYear();
```
I suggest we dont add the year to the title, instead use as title "JSON Schema Roadmap"

benjagm commented on pages/overview/roadmap/index.page.tsx:
```diff
@@ -0,0 +1,128 @@
+      <Headline1>
+        {newTitle} {date}
+      </Headline1>
+
```
```suggestion
      <p>
        Our "Roadmap" section displays our key objectives for the long term. While this roadmap provides a detailed outlook for the near future, please note that it might be subject to change. In fact, we are currently [discussing](https://github.com/orgs/json-schema-org/discussions/813) the new priorities for the next 24-month cycle, which will lead to relevant changes. Please consider joining the discussion to become an active part of JSON Schema's future!
      </p>
```

benjagm reviewed (changes requested):
We are getting very close!!

After reviewing the changes we made after my suggestion I realized that there is a better approach by following best practices to avoid Tailwind CSS purge. Check my comment on that. We are likely going to remove all those changes in the tailwind conf by making sure we don't concatenate string to create class names. https://v2.tailwindcss.com/docs/optimizing-for-production#writing-purgeable-html

In addition, for the mobile version it seems we have bigger left padding and it will be better if we make sure the content is centered with same x padding.

![Screenshot 2024-10-19 at 11 33 53](https://github.com/user-attachments/assets/47cddcff-f239-43aa-b3ea-ca6c8a69d628)

Great job! We are one step closer

rishabhknowss commented:
making changes :>

rishabhknowss commented:
i think its done

benjagm reviewed (approved):
Amazing job! This is ready to merge!
